core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -308,6 +311,15 @@ object SparkSubmit extends CommandLineUtils {
       RPackageUtils.checkAndBuildRPackage(args.jars, printStream, args.verbose)
     }
 
+    // In client mode, download remote files.
+    if (deployMode == CLIENT) {
```

Comment:
sorry I may not have enough background knowledge, why we only do this for client mode?